### ISSUE TITLE
Add a calendar color legend or prefixes for calendar entries

### ISSUE THREAD

**Issue by languitar** [geier/khal#315]
I have quite a few calendars configured and often cannot remember which color relates to which calendar. Therefore, some kind of legend for the colors, or prefixing of entries with their calendar would be really helpful. probably the easiest solution could be to use the colors in the `printcalendars` command. However, I would better like to see the legend in parallel to the normal calendar and agenda views.

**Comment by geier** [comment]
Good idea. Could you perhaps do a mockup how you imagine that to look?

**Comment by languitar** [comment]
I think using multiple approaches would be good:

1. Colors in `printcalendars`. That should be easy to implement.
1. Adding an option to display the calendar with each entry. Something like this could work:
    ```
    10:00-12:00: Foo bar (calendar_1) ⟳
    12:30-13:00: Group Socializing (calendar_bla) ⟳
    14:00-16:00: Talk XYZ (blubb)
    ```

1. Option to display a legend along with the calendars. I'd propose to just display the colored calendar names with `Legend:` as a title. Either below the calendar widget on the left or below the event list.

**Comment by youngmoney** [comment]
the prefixing of calendars can be done with list and --format, (which I hope we replacing all listing/event viewing with)

**Comment by languitar** [comment]
@youngmoney so this is already implemented now? Can you show an example how to use that?

**Comment by youngmoney** [comment]
it is available in master

I simple example is: `khal list --format "{start} {end} {title} ({calendar})"`

or if you want to only do prefixes you can use any form of python string tempting so:

`khal list --format "{start} {end} {title} ({calendar:.4})"` shows only the first 4 characters of the calendar

right now the list command does not use calendar names correctly (no display names) but this could be added in future, there is also the option to use the calendar color using `{calendar-color}` which will be coming later as well

**Comment by languitar** [comment]
Ok, but that doesn't help for ikhal?

**Comment by youngmoney** [comment]
The hope is that the internals of list will get moved into ikhal as well, I am currently working on getting those internals into all of the other commands and will look into ikhal later

**Comment by geier** [comment]
can now be done with ikhal as well.
